src/main/kotlin/no/nav/tilleggsstonader/sak/utbetaling/simulering/SimuleringSteg.kt

```diff
@@ -0,0 +1,41 @@
+package no.nav.tilleggsstonader.sak.utbetaling.simulering
+
+import no.nav.tilleggsstonader.sak.behandling.domain.BehandlingType
+import no.nav.tilleggsstonader.sak.behandling.domain.Saksbehandling
+import no.nav.tilleggsstonader.sak.behandlingsflyt.BehandlingSteg
+import no.nav.tilleggsstonader.sak.behandlingsflyt.StegType
+import no.nav.tilleggsstonader.sak.vedtak.TypeVedtak
+import no.nav.tilleggsstonader.sak.vedtak.VedtaksresultatService
+import org.springframework.stereotype.Service
+
+@Service
+class SimuleringSteg(
+    val simuleringService: SimuleringService,
+    val vedtaksresultatService: VedtaksresultatService,
+) : BehandlingSteg<Void?> {
+
+    override fun utførSteg(saksbehandling: Saksbehandling, data: Void?) {
+        val resultat = vedtaksresultatService.hentVedtaksresultat(saksbehandling)
+
+        if (skalUtFøreSimulering(saksbehandling, resultat)) {
+            simuleringService.hentOgLagreSimuleringsresultat(saksbehandling)
+        }
+    }
+
+    private fun skalUtFøreSimulering(saksbehandling: Saksbehandling, vedtak: TypeVedtak): Boolean =
+        when (saksbehandling.type) {
+            BehandlingType.REVURDERING -> resultatSkalSimuleres(vedtak)
+            BehandlingType.FØRSTEGANGSBEHANDLING -> false
+            else -> error("Behandlingstype ${saksbehandling.type} støttes ikke i simulering")
+        }
+
+    private fun resultatSkalSimuleres(type: TypeVedtak): Boolean {
+        return when (type) {
+            TypeVedtak.INNVILGELSE -> true
+            TypeVedtak.AVSLAG -> false
+            else -> error("Simulering støtter ikke resultat $type")
```

Comment:
Ser poenget med å legge til `else` med en error her hvis en ny blir lagt til. 
Samtidig, hvis man ikke har den og legger til en ny verdi så må man også ta stilling til den her, i stedet for at det feiler vid en ev revurdering :-)